src/manage.c

```diff
@@ -381,28 +389,9 @@ static void manage_acq()
       acq->dopp_hint_low = cf - ACQ_FULL_CF_STEP;
       acq->dopp_hint_high = cf + ACQ_FULL_CF_STEP;
     }
-    return;
-  }
-
-  /* Transition to tracking. */
-  u32 track_count = nap_timing_count() + 20000;
-  cp = propagate_code_phase(cp, cf, track_count - timer_count);
-
-  // Contrive for the timing strobe to occur at or close to a PRN edge (code phase = 0)
-  track_count += 16*(1023.0-cp)*(1.0 + cf / GPS_L1_HZ);
-
-  /* Start the tracking channel */
-  tracking_channel_init(chan, acq->sid, cf, track_count, cn0,
-                        TRACKING_ELEVATION_UNKNOWN);
-  /* TODO: Initialize elevation from ephemeris if we know it precisely */
-
-  /* Start the decoder channel */
-  if (!decoder_channel_init(chan, acq->sid)) {
-    log_error("decoder channel init failed");
+  } else {
+    acq->state = ACQ_PRN_TRACKING;
```

Comment:
Isn't it better to not remember on acquisition side which SVs are being tracked and just query that from tracking in choose_acq_sat function (that query won't require heavy thread synchronization as it's about atomic reading of status field)? Then here acquisition can just blindly form a tracking_startup_params_t structure and throw it to tracking. If we put tracking_startup_params_t structure as a field to acq_status_t and add a link to the next tracking_startup_params_t we can use the same technique as in other places to form a queue of requests to tracking. This would improve inter-thread communication IMO as won't be stopping threads for performing potentially long operations (like tracking start).